### ISSUE TITLE
ant-design-pro-template运行npm run test报错

### ISSUE THREAD

**Issue by lys623** [ant-design/ant-design-pro#776]
![image](https://user-images.githubusercontent.com/4364314/35179432-2388c6bc-fdd5-11e7-85f2-426fbc94e1d5.png)

**Comment by chenshuai2144** [comment]
应该是少了一个运行库

**Comment by jwmann** [comment]
I have the same problem.

Steps to reproduce.
````
$> npm i -g ant-design-pro-cli
$> pro new
$> cd demo
$> npm test
````

**Comment by jwmann** [comment]
Jest also fails

![image](https://user-images.githubusercontent.com/436270/35241302-0725645c-ff84-11e7-9ca4-bf070c5ea44d.png)

**Comment by jwmann** [comment]
I thought perhaps the issue might have something to do with `roadhog@2.2.0` but the error happens for `roadhog@2.1.0` as well. 😭

**Comment by jwmann** [comment]
Further testing shows that it's failing at this node module path
````
/node_modules/umi-test/lib/transformers/jsTransformer.js
````

`roadhog` is looking for `umi-test@^0.3.0` however a fresh install will install `umi-test@^0.3.1`
If you manually install `npm i umi-test@^0.3.0` you get different error messages. The same errors that Jest above is giving.

**Comment by jwmann** [comment]
Was #640 merged without testing? Every module warning seems to related to the removed modules in this commit

**Comment by jwmann** [comment]
So `umi-test@^0.3.1` changes the way it exports its module in its `jsTransformer.js`
By doing that, `jest-runtime` is looking for a `process` function which because `umi-test` isn't exported properly, it will fail.

jest-runtime sees this transform object:
````
{ 
  default: { 
    canInstrument: true,
    getCacheKey: [Function: getCacheKey],
    process: [Function: process] 
  } 
}
````

But `umi-test` script_transformer.js is checking for `transform.process` not `transform.default.process`
Fixing either the above or umi-test will make it work but it will still result in the errors of missing modules.

`umi-test@0.3.1` breaks
Fix by doing `npm i umi-test@0.3.0`

Now missing module errors.

**Comment by chenshuai2144** [comment]
I can not reproduce it 。。。。。

**Comment by jwmann** [comment]
What version is your `/node_modules/umi-test`?
At the very least that shouldn't be working, the author of the module has confirmed the bug.

**Comment by jwmann** [comment]
Now that `umi-test` has reverted `0.3.1` to `0.3.0`

I decided to start fresh.
The tests don't have the same error, some of them seem to pass although with warnings.

Still fails from fresh install.
![image](https://user-images.githubusercontent.com/436270/35341235-2c5f9272-00f3-11e8-9e63-4dcaab2d15e6.png)

**Comment by chenshuai2144** [comment]
关注中

**Comment by jwmann** [comment]
Yes very. I can't use this software unless these tests are working at the very least from the demo

**Comment by jwmann** [comment]
Alright, so I think the  issue if that my computer is too slow to actually run these tests in a reasonable timeframe.

Also we aren't passing the `done()` function into the `it()` tests.
Source: https://stackoverflow.com/a/25273095/185731

These are my _modified_ tests:

#### login.e2e.js
````
import Nightmare from 'nightmare';

describe('Login', () => {
  jasmine.DEFAULT_TIMEOUT_INTERVAL = 300000; // 5 minute timeout
  let page;
  beforeEach(() => {
    page = Nightmare();
    page
      .goto('http://localhost:8000/')
      .evaluate(() => {
        window.localStorage.setItem('antd-pro-authority', 'guest');
      })
      .goto('http://localhost:8000/#/user/login');
  });

  it('should login with failure', async (done) => {
    await page.type('#userName', 'mockuser')
      .type('#password', 'wrong_password')
      .click('button[type="submit"]')
      .wait('.ant-alert-error') // should display error
      .end();
    done();
  });

  it('should login successfully', async (done) => {
    const text = await page.type('#userName', 'admin')
      .type('#password', '888888')
      .click('button[type="submit"]')
      .wait('.ant-layout-sider h1') // should display error
      .evaluate(() => document.body.innerHTML)
      .end();
    expect(text).toContain('<h1>Ant Design Pro</h1>');
    done();
  });
});
````

#### home.e2e.js
````
import Nightmare from 'nightmare';

describe('Homepage', () => {
  jasmine.DEFAULT_TIMEOUT_INTERVAL = 300000; // 5 minute timeout
  it('it should have logo text', async (done) => {
    const page = Nightmare().goto('http://localhost:8000');
    const text = await page.wait('h1').evaluate(() => document.body.innerHTML).end();
    expect(text).toContain('<h1>Ant Design Pro</h1>');
    done();
  });
});
````

**Comment by jwmann** [comment]
![image](https://user-images.githubusercontent.com/436270/35349296-91fe0d2e-0108-11e8-9c7a-113708afb9ce.png)

**Comment by chenshuai2144** [comment]
fix in #1006
